File: reactapp/src/types/substitution/conceptual-knowledge/substitutionEquation.ts
import { Type } from "class-transformer";
import { LinearEquation } from "@/types/math/linearEquation.ts";

export interface ISubstitutionEquation {
    readonly equation: LinearEquation;
    readonly isIsolated: boolean;
}

export class SubstitutionEquation implements ISubstitutionEquation {
    @Type(() => LinearEquation)
    public readonly equation: LinearEquation;

    public readonly isIsolated: boolean;

    constructor(equation: LinearEquation, isIsolated: boolean) {
        this.equation = equation;
        this.isIsolated = isIsolated;
    }
}

File: webapi/Models/Substitution/ConceptualKnowledge/SubstitutionEquation.cs
using webapi.Models.Math;

namespace webapi.Models.Substitution.ConceptualKnowledge
{
    public class SubstitutionEquation
    {
        public required LinearEquation Equation { get; set; }

        public bool IsIsolated { get; set; }

        public SubstitutionEquation() { }
    }
}
